ddl/generated_column.go

```diff
@@ -267,6 +271,27 @@ func checkIllegalFn4GeneratedColumn(colName string, expr ast.ExprNode) error {
 	if c.hasAggFunc {
 		return ErrInvalidGroupFuncUse
 	}
+	if c.hasRowVal {
+		return ErrGeneratedColumnRowValueIsNotAllowed.GenWithStackByArgs(colName)
+	}
+	return nil
+}
+
+func checkIllegalFn4GeneratedIndex(idxName string, expr ast.ExprNode) error {
```

Comment:
It's better to combine two functions into one. Their logics are same, except for the error message.

Reply:
done